hs_app_netCDF/tests/views/test_create_netcdf_resource.py

```diff
@@ -0,0 +1,135 @@
+        post_data = {'resource-type': 'NetcdfResource',
+                     'title': 'Test NetCDF Resource Creation',
+                     'irods_federated': 'true'
+                     }
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
+        hydroshare.delete_resource(res_id)
+        self.assertEqual(BaseResource.objects.count(), 0)
+
+        # test with file upload
+        self.assertEqual(ResourceFile.objects.count(), 0)
+        post_data = {'resource-type': 'NetcdfResource',
+                     'title': 'Test NetCDF Resource Creation',
+                     'irods_federated': 'true',
+                     'files': (self.netcdf_file_name, open(self.netcdf_file))
+                     }
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
+        # there should be 2 files (nc and txt) - successful metadata extraction
+        self.assertEqual(ResourceFile.objects.count(), 2)
+
+        hydroshare.delete_resource(res_id)
+
+    def test_create_resource_with_invalid_file(self):
+        # here we are testing the create_resource view function
+
+        self.assertEqual(BaseResource.objects.count(), 0)
+        self.assertEqual(ResourceFile.objects.count(), 0)
+        # test with bad nc file - this file should not be uploaded and the resource
+        # is also not created
+        post_data = {'resource-type': 'NetcdfResource',
+                     'title': 'Test NetCDF Resource Creation',
+                     'irods_federated': 'true',
+                     'files': (self.netcdf_bad_file_name,
+                               open(self.netcdf_bad_file))
+                     }
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        # this should take back to create_resource.html page as resource was not created
+        self.assertIn('<!doctype html>', response.content)
```

Comment:
Perhaps in addition to this assertIn(), adding a more create-resource page specific assertIn() is better such as ```self.assertIn('create-resource', response.content)```?